### ISSUE TITLE
Rework HTTP Client Configuration support

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#41138]
We have a number of related issues that we can't easily fix until we redesign the way that HTTP Clients are configured. See `theme: http-client-config`.

**Comment by anmol797** [comment]
Hi , can i also contribute here

**Comment by wilkinsona** [comment]
Thanks for the offer, @anmol797, but as indicated by the `status: pending-design-work` label, we need to do some design work here before we can consider accepting contributions.

**Comment by wilkinsona** [comment]
The theme's gone now so I'm guessing that this is no longer needed.
